### PR TITLE
allow solvability if feedstock wants it

### LINKED ISSUE #1031: Enable solvability for version bumps as an option
This would enable projects that have strict version bump orders (KDE, boto, etc.) to enable solvability checks to make certain that PRs aren't opened that fail on arrival.

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #1031 
@scopatz

### CODE REVIEW

File: conda_forge_tick/auto_tick.py [hunk 1]
```diff
@@ -225,7 +224,13 @@ def run(
                 )
             ]
 
-    if migrator.check_solvable and not is_recipe_solvable(feedstock_dir):
+    if (
+        migrator.check_solvable
```
[the remark this answers is not in the document]
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:228]: What parens are missing? There are parens around `check_solvable` and the `conda-forge.yml` inspection.

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:228]: black keeps removing any extra parens

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:228]: I take it back, I fixed the parens

Comment (beckermr) [inline conda_forge_tick/auto_tick.py:228]:
```suggestion
        (
            migrator.check_solvable
            or feedstock_ctx.attrs["conda-forge.yml"]
            .get("bot", {})
            .get("check_solvability", False)
        )
```

Comment (beckermr) [reply conda_forge_tick/auto_tick.py:228]: You should group the or statement together for clarity.

Comment (beckermr) [reply conda_forge_tick/auto_tick.py:228]: Yup you got it!

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:228]: Ok, I've updated that, does it now reflect what you wanted?

Comment (beckermr) [reply conda_forge_tick/auto_tick.py:228]: ignore my suggestion

File: conda_forge_tick/auto_tick.py [hunk 2]
```diff
@@ -225,7 +224,12 @@ def run(
...
+        or feedstock_ctx.attrs["conda-forge.yml"]
+        .get("bot", {})
+        .get("check_solvability", False)
```
Comment (beckermr) [inline conda_forge_tick/auto_tick.py:231]: We use "check_solvable" in the migrator yaml. We should use that here.

https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/migrations/pypy.yaml#L15

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:231]: We are that is on line 228. This is checking for if the *feedstock* wants to check solvability, regardless of the migrator (eg. the boto stack wants to check solvability for version bumps)

Comment (beckermr) [reply conda_forge_tick/auto_tick.py:231]: I mean the key in the yaml. It should be named the same. I don’t want to have to remember which one goes where.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**